calico_containers/calico_ctl/profile.py

```diff
@@ -20,8 +20,8 @@
   calicoctl profile <PROFILE> tag (add|remove) <TAG>
   calicoctl profile <PROFILE> rule add (inbound|outbound) [--at=<POSITION>]
     (allow|deny) [(
-      (tcp|udp) [(from [(ports <SRCPORTS>)] [(tag <SRCTAG>)] [<SRCCIDR>])]
-                [(to   [(ports <DSTPORTS>)] [(tag <DSTTAG>)] [<DSTCIDR>])] |
+      (tcp|udp) [(from [(ports <SRCPORTS>)] [(tag <SRCTAG>)] [(cidr <SRCCIDR>)])]
```

Comment:
Unless I am mistaken, this feels like a bug in the doc opt parser rather than in our definitions.  Out of interest did you have a look to see if doc opt could be fixed?

Anyways... assuming we go with this interface change (which you should just run by the team), we should add  the cidr to the other similar branches (e.g. icmp, <no ip type> and the profile rule remove command).  It seems extremely odd to just have it on the tcp|udp branch.  I'd also be surprised if we don't hit the same error for those other branches.  Did you try that out?

We have the ability to show the rules in pretty format - this should match the format used to enter them on the command line - so there is a missing code change there too.  I think it is done in the Rules class in the datastore_datatypes class.